ui/src/app/shared/services/archived-workflows-service.ts

```diff
@@ -39,12 +39,12 @@ export class ArchivedWorkflowsService {
         }
     }
 
-    public listLabelValues(key: string, namespace: string) {
-        if (namespace === '') {
-            return requests.get(`api/v1/archived-workflows-label-values?listOptions.labelSelector=${key}`).then(res => res.body as models.Labels);
-        } else {
-            return requests.get(`api/v1/archived-workflows-label-values?namespace=${namespace}&listOptions.labelSelector=${key}`).then(res => res.body as models.Labels);
+    public async listLabelValues(key: string, namespace: string): Promise<models.Labels> {
+        let url = `api/v1/archived-workflows-label-values?listOptions.labelSelector=${key}`;
+        if (namespace !== '') {
+            url += `&namespace=${namespace}`;
         }
+        return (await requests.get(url)).body as models.Labels;
     }
```

Comment:
that's cleaner :)

Reply:
Aa, I see. Just felt that this
```ts
    private async fetchArchivedWorkflowsLabels(key: string): Promise<string[]> {
        const labels = await services.archivedWorkflows.listLabelValues(key, this.props?.namespace);
        return labels.items
            .map(i => key + '=' + i)
            .sort((a, b) => a.localeCompare(b));
    }
```
is easier to read than this
```ts
    private async fetchArchivedWorkflowsLabels(key: string): Promise<string[]> {
        const labels = await services.archivedWorkflows.listLabelValues(key, this.props?.namespace);
        return labels.items.map(i => key + '=' + i).sort((a, b) => a.localeCompare(b));
    }
```
I guess the linter disagrees ^^;

Reply:
I’m with you there…go figure